app/controllers/visits_controller.rb

```diff
@@ -42,6 +42,12 @@ def update
 
     if @visit.update_attributes(visit_params)
       @visit.sub_service_request.set_to_draft unless @in_admin
+      if visit_params[:set_all_insurance] == '1'
+        insurance_qty = visit_params[:insurance_billing_qty].to_i
+        @visit_group.visits.each do |visit|
```

Comment:
You can use `update_all` here to update all of the records in a single query for performance